src/iuse_actor.cpp

```diff
@@ -284,6 +284,57 @@ void iuse_transform::info( const item &it, std::vector<iteminfo> &dump ) const
     }
 }
 
+std::unique_ptr<iuse_actor> unpack_actor::clone() const
+{
+    return std::make_unique<unpack_actor>( *this );
+}
+
+void unpack_actor::load( const JsonObject &obj )
+{
+    obj.read( "group", unpack_group );
+    obj.read( "items_fit", items_fit );
+    assign( obj, "filthy_volume_threshold", filthy_vol_threshold );
+}
+
+int unpack_actor::use( player &p, item &it, bool, const tripoint & ) const
+{
+    std::vector<item> items = item_group::items_from( unpack_group, calendar::turn );
+    item last_armor;
+
+    p.add_msg_if_player( _( "You unpack the %s." ), it.tname() );
+
+    for( item &content : items ) {
+        if( content.is_armor() ) {
+            if( items_fit && !content.has_flag( flag_FIT ) ) {
+                content.set_flag( flag_FIT );
```

Comment:
Check for `!content.has_flag( flag_FIT )` is redundant here and below on line 311. If `content` doesn't have the flag, it will be set. If `content` already has this flag, setting it once again will do nothing.

The same for `content.has_flag( flag_FIT )` below on line 313. Trying to unset flag from an item which doesn't have the said flag will do nothing.